### PR TITLE
Prop types

### PR DESCRIPTION
**Pull request by SamuelHarner**
Currently trying out prop-types to see if it would be good to use going forward. My current thoughts are it could be a good tool for implementing some type-checking and could serve as useful documentation on what types should be provided to a component.

However, so far I think most of the eslint prop-type errors that we have in navigation components may kind of be false-flags, that do not really require restructuring or use of prop-types, as we have followed the official documentation for the react-navigation components and in many cases props are generated as a byproduct of creating a component (making it unclear how/unnecessary to place restrictions on those props). See some commits below for examples.

### CODE REVIEW

**Review by SamuelHarner — COMMENTED** [review]
Think the best option may be to just silence eslint in this case, based on discussion here: https://stackoverflow.com/questions/57232105/where-should-proptypes-for-navigationoptions-inside-createbottomtabnavigator-be

Although it is possible to restructure how `tabBarIcon` is passed to `iconProperties` to enable explicit prop-type declaration of the nested props, this solution would still probably require prop-spreading; `{...props}`, which would also generate a linting error.

File: App/components/BottomNavigation.jsx [hunk 1]
```diff
@@ -36,6 +36,7 @@ const BottomNavigation = () => {
 };
 
 const iconProperties = ({ route }) => ({
+  // eslint-disable-next-line react/prop-types
```
Comment (SamuelHarner) [inline App/components/BottomNavigation.jsx:39]: Think the best option may be to just silence eslint in this case, based on discussion here: https://stackoverflow.com/questions/57232105/where-should-proptypes-for-navigationoptions-inside-createbottomtabnavigator-be

Although it is possible to restructure how `tabBarIcon` is passed to `iconProperties` to enable explicit prop-type declaration of the nested props, this solution would still probably require prop-spreading; `{...props}`, which would also generate a linting error.

**Comment by jscarlsson** [comment]
I think I agree that it makes sense to silence it in BottomNavigation.jsx. However that seems annoying to have to do every time, so maybe we should just disable it altogether. That said, the same error also occurs in TrackingNavigation.jsx, and here it would be pretty easy to just do the checking. In theory I think it makes sense to want to do it, but I don't think in our case it adds much value for the extra hassle.

I will say that I don't fully understand prop-spreading (maybe I should read a bit more) but it seems like a thing that we might want at least sometimes. So maybe it should be a warning instead of an error. Also, we don't actually have to use it (right? you could just write them out?) but that just seems more annoying. But point is I don't think that is a problem in and of itself.

So at the end of the day, I don't actually know what to do.

Some more (basic) info:
https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/prop-types.md
https://reactjs.org/docs/typechecking-with-proptypes.html
And of course (though it didn't feel very helpful): https://github.com/airbnb/javascript/tree/master/react

**Comment by SamuelHarner** [comment]
@jscarlsson and I have come to the conclusion that the prop linting rules are not necessary to follow in our current components, which only serve the purpose of navigation. 

PropTypes may however be useful going forward when we create more of our own components with varying functionality, in which case PropTypes will provide documentation on how to properly use these components.

So it would be good if PropTypes are used for new components when appropriate, but otherwise the linting rule can just be disabled as we have done here for navigation components.

**Review by jscarlsson — APPROVED** [review]
Looks good :+1: 

I did a separate PR (#47) where I also changed stuff regarding bools, but I took the exact opposite approach. Maybe yours is better, though. Either way, this is fine to merge, and maybe we'll talk about how to handle that (as a group if anybody actually cares). It's a really minor thing at the end of the day.

**Comment by jscarlsson** [comment]
oh shit you changed it again already

well, again, I'd say it's fine to merge either way

**Comment by SamuelHarner** [comment]
Haha sorry, saw your boolean-value changes and reset mine. Thought it was a better idea to fix other linting stuff in a dedicated PR like you did, also think your approach to follow the default is just as good if not better so go with that I would say!

Cool, merging this PR 👍

**Pull request merged**